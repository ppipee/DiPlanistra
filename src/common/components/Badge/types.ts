export enum BadgeVariant {
	Outlined = 'outlined',
	Filled = 'filled',
}

export type BadgeVariantTypes = BadgeVariant | 'filled' | 'outlined'

export interface BadgeProps {
	$variant?: BadgeVariantTypes
	$color: string
	$secondColor?: string
}
