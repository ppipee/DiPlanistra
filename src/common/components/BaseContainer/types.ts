import { ScreenName } from 'common/styles/utils/viewport/screenSizes'

export interface BaseContainerProps {
	$screen?: ScreenName
	$padding?: string
	$paddingMobile?: string
	$margin?: string
	$marginMobile?: string
}
