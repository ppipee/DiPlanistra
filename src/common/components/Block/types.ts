export interface BlockProps {
	$padding?: string
	$borderRadius?: string
	$margin?: string
}
