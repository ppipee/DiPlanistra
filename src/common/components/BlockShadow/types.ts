export enum Shadow {
	Default = 'default',
	Hard = 'hard',
}

type ShadowType = Shadow | Shadow.Default | Shadow.Hard

export interface BlockShadowProps {
	$variant?: ShadowType
}
