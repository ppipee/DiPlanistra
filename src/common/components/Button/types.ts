export interface ButtonProps {
	$size?: SizeType
	$variant?: VariantType
	$responsive?: boolean
	$disable?: boolean
	$color: string
	$secondaryColor?: string
	$shadow?: boolean
	$border?: BorderType
}

type SizeType = ButtonSizes | 'small' | 'default'

export enum ButtonSizes {
	Small = 'small',
	Default = 'default',
}

type VariantType = ButtonVariants | 'outlined' | 'filled'

export enum ButtonVariants {
	Outlined = 'outlined',
	Filled = 'filled',
}

type BorderType = ButtonBorders | 'default' | 'curve'

export enum ButtonBorders {
	Default = 'default',
	Curve = 'curve',
}
