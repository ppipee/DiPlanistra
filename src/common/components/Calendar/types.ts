export interface CalenderDateProps {
	startDate?: Date
	endDate?: Date
}

export interface CalendarDisplayStateProps {
	year: number
	month: number
}

export interface CalendarDisplayRange {
	startRange?: CalendarDisplayStateProps
	endRange?: CalendarDisplayStateProps
}

export interface CalendarCellProps {
	$inBetween?: boolean
	$selected?: boolean
	$variant?: CalendarCellVariant
	$isFirstCell?: boolean
	$isLastCell?: boolean
	$isDateSelectable?: boolean
}

export type CalendarCellVariant = 'currentMonth' | 'otherMonth' | 'currentDate'
