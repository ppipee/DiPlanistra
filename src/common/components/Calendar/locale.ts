export const CALENDAR_START_DATE_TITLE = ['เริ่มต้น', 'Start Date']
export const CALENDAR_END_DATE_TITLE = ['สิ้นสุด', 'End Date']
