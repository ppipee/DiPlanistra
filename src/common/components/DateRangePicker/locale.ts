export const PICKER_YESTERDAY = ['เมื่อวาน', 'Yesterday']
export const PICKER_LAST_7_DAYS = ['7 วันที่แล้ว', 'Last 7 Days']
export const PICKER_LAST_14_DAYS = ['14 วันที่แล้ว', 'Last 14 Days']
export const PICKER_LAST_30_DAYS = ['30 วันที่แล้ว', 'Last 30 Days']
export const PICKER_THIS_MONTH = ['เดือนนี้', 'This Month']
export const CUSTOM_PICKER = ['กำหนดเอง', 'Custom']
