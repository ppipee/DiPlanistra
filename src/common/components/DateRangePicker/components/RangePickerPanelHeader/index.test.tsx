import React from 'react'

import 'jest-styled-components'
import { render } from '@testing-library/react'

import { createI18nWrapper } from 'core/locale/testUtils'
import { Locale } from 'core/locale/types'

import * as RangePickerPanelHeaderType from '.'

describe('Test RangePickerPanelHeader', () => {
	const useDateRangeContextSpy = jest.fn()
	const baseDateInputSpy = jest.fn()

	jest.doMock('common/components/DateRangePicker/context', () => ({
		useDateRangeContext: useDateRangeContextSpy,
	}))

	jest.doMock('common/components/DateRangePicker/components/BaseDateInput', () => (props: any) => {
		baseDateInputSpy(props)

		return <div {...props} />
	})

	jest.doMock('common/components/Gap', () => (props: any) => <div {...props}>{props.children}</div>)

	const { default: RangePickerPanelHeader } = require('.') as typeof RangePickerPanelHeaderType

	beforeEach(() => {
		useDateRangeContextSpy.mockReturnValue({
			date: { startDate: new Date(2020, 0, 1), endDate: new Date(2020, 0, 2) },
		})
	})

	afterEach(() => {
		jest.resetAllMocks()
	})

	it('should render RangePickerPanelHeader with TH text correctly', () => {
		const { container } = render(<RangePickerPanelHeader />, {
			wrapper: createI18nWrapper(Locale.th),
		})

		expect(useDateRangeContextSpy).toBeCalledTimes(1)
		expect(baseDateInputSpy.mock.calls[0][0].value).toBe('01/01/2020')
		expect(baseDateInputSpy.mock.calls[1][0].value).toBe('02/01/2020')
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render RangePickerPanelHeader with EN text correctly', () => {
		render(<RangePickerPanelHeader />, {
			wrapper: createI18nWrapper(Locale.en),
		})

		expect(useDateRangeContextSpy).toBeCalledTimes(1)
		expect(baseDateInputSpy.mock.calls[0][0].value).toBe('01/01/2020')
		expect(baseDateInputSpy.mock.calls[1][0].value).toBe('02/01/2020')
	})

	it('should not set empty string if start date is undefined', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: { endDate: new Date(2020, 0, 2) },
		})

		render(<RangePickerPanelHeader />, {
			wrapper: createI18nWrapper(Locale.en),
		})

		expect(useDateRangeContextSpy).toBeCalledTimes(1)
		expect(baseDateInputSpy.mock.calls[0][0].value).toBe('')
		expect(baseDateInputSpy.mock.calls[1][0].value).toBe('02/01/2020')
	})

	it('should not set empty string if end date is undefined', () => {
		useDateRangeContextSpy.mockReturnValue({
			date: { startDate: new Date(2020, 0, 1) },
		})

		render(<RangePickerPanelHeader />, {
			wrapper: createI18nWrapper(Locale.en),
		})

		expect(useDateRangeContextSpy).toBeCalledTimes(1)
		expect(baseDateInputSpy.mock.calls[0][0].value).toBe('01/01/2020')
		expect(baseDateInputSpy.mock.calls[1][0].value).toBe('')
	})
})
