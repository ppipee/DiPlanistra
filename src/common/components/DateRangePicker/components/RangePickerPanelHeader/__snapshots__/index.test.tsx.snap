// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RangePickerPanelHeader should render RangePickerPanelHeader with TH text correctly 1`] = `
.c0 input,
.c0 svg {
  pointer-events: none;
}

.c0 input:focus {
  border: 1px solid #D3D6DA;
  box-shadow: none;
}

<div>
  <div
    class="c0"
  >
    <div
      label="เริ่มต้น"
      value="01/01/2020"
    />
  </div>
  <div
    class="c0"
  >
    <div
      label="สิ้นสุด"
      value="02/01/2020"
    />
  </div>
</div>
`;
